job/server/src/main/java/alluxio/job/plan/migrate/MigrateDefinition.java

```diff
@@ -160,17 +164,26 @@ private static void migrate(MigrateCommand command, WritePType writeType, FileSy
         CreateFilePOptions.newBuilder().setWriteType(writeType).build();
     OpenFilePOptions openFileOptions =
         OpenFilePOptions.newBuilder().setReadType(ReadPType.NO_CACHE).build();
-    try (FileInStream in = fileSystem.openFile(new AlluxioURI(source), openFileOptions);
-         FileOutStream out = fileSystem.createFile(new AlluxioURI(destination), createOptions)) {
-      try {
-        IOUtils.copyLarge(in, out, new byte[8 * Constants.MB]);
-      } catch (Throwable t) {
+    final AlluxioURI destinationURI = new AlluxioURI(destination);
+    try {
+      try (FileInStream in = fileSystem.openFile(new AlluxioURI(source), openFileOptions);
+           FileOutStream out = fileSystem.createFile(destinationURI, createOptions)) {
         try {
-          out.cancel();
-        } catch (Throwable t2) {
-          t.addSuppressed(t2);
+          IOUtils.copyLarge(in, out, new byte[8 * Constants.MB]);
+        } catch (Throwable t) {
+          try {
+            out.cancel();
+          } catch (Throwable t2) {
+            t.addSuppressed(t2);
+          }
+          throw t;
         }
-        throw t;
+      }
+    } catch (FileAlreadyExistsException e) {
+      if (overwrite) {
+        fileSystem.delete(destinationURI);
```

Comment:
@bradyoo a couple of high level questions.  if this does not throw an exception, would it be considered successful and not retried? also if a user specified overwrite, maybe we should delete and then copy and not rely on a retry?

Reply:
Originally this code was the intuitive version of 

if (exists) { delete } but then I did a couple experiments and it appears that for small files, it's slower than doing this.

Reply:
i see.  So if we look at MigrateDefinition by itself, it does not actually overwrite any files. Its caller is causing the overwrite behavior, by itself it simply deletes it. 

Does it make sense to have a retry within the MigrateDefinition code, so that any (future/ other) caller of MigrateDefinition will get the overwrite behavior?

Reply:
Oh nevermind I see the bug. fixing.